torch/csrc/jit/codegen/cuda/lower_alias_memory.cpp

```diff
@@ -356,8 +368,134 @@ struct ScopeInfo {
   kir::ForLoop* loop = nullptr;
 };
 
-using ScopeInfoOwningPtr = std::unique_ptr<ScopeInfo>;
-using ScopeInfoOwningPtrList = std::vector<ScopeInfoOwningPtr>;
+//! Assign an integer position to each expression to help representing
+//! scope ranges. The position starts from 1.
+class ExprPosMap {
+ public:
+  //! Get the position of an expr
+  int get(const Expr* expr) const {
+    auto it = expr_pos_map_.find(expr);
+    TORCH_INTERNAL_ASSERT(it != expr_pos_map_.end());
+    return it->second;
+  }
```

Comment:
nit: I think this can be simplified as:
```suggestion
  int get(const Expr* expr) const {
    return expr_pos_map_.at(expr);
  }
```